ts-client/src/ValorClient.ts

```diff
@@ -0,0 +1,334 @@
+   * Creates a new model
+   *
+   * @param name name of the model
+   * @param metadata metadata of the model
+   *
+   * @returns {Promise<void>}
+   */
+  public async createModel(name: string, metadata: object): Promise<void> {
+    await this.client.post('/models', { name, metadata });
+  }
+
+  /**
+   * Deletes a model
+   *
+   * @param name name of the model to delete
+   *
+   * @returns {Promise<void>}
+   */
+  public async deleteModel(name: string): Promise<void> {
+    await this.client.delete(`/models/${name}`);
+  }
+
+  /**
+   * Creates a new evaluation or gets an existing one if an evaluation with the
+   * same parameters already exists.
+   *
+   * @param model name of the model
+   * @param dataset name of the dataset
+   * @param taskType type of task
+   *
+   * @returns {Promise<Evaluation>}
+   */
+  public async createOrGetEvaluation(
+    model: string,
+    dataset: string,
+    taskType: TaskType
+  ): Promise<Evaluation> {
+    const response = await this.client.post('/evaluations', {
+      model_names: [model],
+      datum_filter: { dataset_names: [dataset] },
+      parameters: { task_type: taskType }
+    });
+    return response.data[0];
+  }
+
+  /**
+   * Fetches evaluations matching the filters defined by queryParams. This is
+   * private because we define higher-level methods that use this.
+   *
+   * @param queryParams An object containing query parameters to filter evaluations by.
+   *
+   * @returns {Promise<Evaluation[]>}
+   */
+  private async getEvaluations(queryParams: object): Promise<Evaluation[]> {
+    const response = await this.client.get('/evaluations', { params: queryParams });
+    return response.data;
+  }
+
+  /**
+   * Fetches an evaluation by id
+   *
+   * @param id id of the evaluation
+   *
+   * @returns {Promise<Evaluation>}
+   */
+  public async getEvaluationById(id: number): Promise<Evaluation> {
+    const evaluations = await this.getEvaluations({ evaluation_ids: id });
+    return evaluations[0];
+  }
+
+  /**
+   * Fetches all evaluations associated to a given model
+   *
+   * @param modelName name of the model
+   *
+   * @returns {Promise<Evaluation[]>}
+   */
+  public async getEvaluationsByModelName(modelName: string): Promise<Evaluation[]> {
+    return this.getEvaluations({ models: modelName });
+  }
+
+  /**
+   * Fetches all evaluations associated to a given dataset
+   *
+   * @param datasetName name of the dataset
+   *
+   * @returns {Promise<Evaluation[]>}
+   */
+  public async getEvaluationsByDatasetName(datasetName: string): Promise<Evaluation[]> {
+    return this.getEvaluations({ datasets: datasetName });
+  }
+
+  /**
+   * Adds ground truth annotations to a dataset
+   *
+   * @param datasetName name of the dataset
+   * @param datumUid uid of the datum
+   * @param annotations annotations to add
+   *
+   * @returns {Promise<void>}
+   */
+  public async addGroundTruth(
+    datasetName: string,
+    datumUid: string,
+    annotations: object[]
```

Comment:
we dont have more specific typing we can use here?

Reply:
they are but its just a tone of types that it requires. i originally had these https://github.com/Striveworks/valor/blob/659952aa9baf7cb14a5d2da1cb8ae283f0e17eb2/ts-client/src/ValorClient.ts but got rid of them to simplify since this method is only used for testing purposes-- i don't think there will be any case (at least soon) where the UI will be creating groundtruth. but happy to add them back if you think that makes sense